### PR TITLE
dev deps: Add strict version contraints & CI: build gem with strict mode & CI: Use Ruby 3.3 for releases

### PR DESCRIPTION
**Pull request by bastelfreak**
also contains #524

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,12 +11,12 @@ jobs:
     if: github.repository_owner == 'voxpupuli'
     steps:
       - uses: actions/checkout@v4
-      - name: Install Ruby 3.1
+      - name: Install Ruby 3.3
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: '3.1'
+          ruby-version: '3.3'
```
Comment (ekohl) [inline .github/workflows/release.yml:17]: In another place I've been lazy and just used Ruby 3. Is that something we want here as well?
```suggestion
          ruby-version: '3'
```

File: json-schema.gemspec [hunk 2]
```diff
@@ -16,13 +15,11 @@ Gem::Specification.new do |s|
   s.extra_rdoc_files = ['README.md', 'LICENSE.md']
   s.required_ruby_version = '>= 2.7'
   s.license = 'MIT'
-  s.required_rubygems_version = '>= 2.5'
 
-  s.add_development_dependency 'bundler'
   s.add_development_dependency 'minitest', '~> 5.0'
-  s.add_development_dependency 'rake'
+  s.add_development_dependency 'rake', '~> 13.2', '>= 13.2.1'
```
Comment (ekohl) [inline json-schema.gemspec:20]: Why so specific? I'd imagine this works just as well
```suggestion
  s.add_development_dependency 'rake', '~> 13.0'
```

File: json-schema.gemspec [hunk 3]
```diff
@@ -16,13 +15,11 @@ Gem::Specification.new do |s|
...
   s.add_development_dependency 'voxpupuli-rubocop', '~> 2.8.0'
-  s.add_development_dependency 'webmock'
+  s.add_development_dependency 'webmock', '~> 3.23', '>= 3.23.1'
```
Comment (ekohl) [inline json-schema.gemspec:22]: Same here:
```suggestion
  s.add_development_dependency 'webmock', '~> 3.23'
```

File: json-schema.gemspec [hunk 4]
```diff
@@ -16,13 +15,11 @@ Gem::Specification.new do |s|
...
 
-  s.add_runtime_dependency 'addressable', '>= 2.8'
+  s.add_runtime_dependency 'addressable', '~> 2.8', '>= 2.8.7'
```
Comment (ekohl) [inline json-schema.gemspec:24]:
```suggestion
  s.add_runtime_dependency 'addressable', '~> 2.8'
```

File: .github/workflows/release.yml (hunk 1 above)
Comment (bastelfreak) [reply .github/workflows/release.yml:17]: I'm not sure. I don't know under which conditions they will bump that version. I only bump the version in the release action when we've the same version in a test and it passes.

**Review by TheMeier — APPROVED** [review]

**Pull request merged**